node/src/bin/manager.rs

```diff
@@ -1165,16 +1179,37 @@ async fn main() -> anyhow::Result<()> {
             commands::assign::reassign(ctx.primary_pool(), &sender, &deployment, node)
         }
         Pause { deployment } => {
+            let ctx_clone = ctx.clone();
+
             let sender = ctx.notification_sender();
-            commands::assign::pause_or_resume(ctx.primary_pool(), &sender, &deployment, true)
+            let pool = ctx.primary_pool();
+            commands::assign::pause_or_resume(
+                ctx_clone.primary_pool(),
+                &sender,
+                &deployment.locate_unique(&pool).unwrap(),
+                true,
+            )
         }
         Resume { deployment } => {
             let sender = ctx.notification_sender();
-            commands::assign::pause_or_resume(ctx.primary_pool(), &sender, &deployment, false)
+            let pool = ctx.primary_pool();
+            commands::assign::pause_or_resume(
+                pool.clone(),
+                &sender,
+                &deployment.locate_unique(&pool).unwrap(),
+                false,
+            )
         }
         Restart { deployment, sleep } => {
+            let ctx_clone = ctx.clone();
             let sender = ctx.notification_sender();
-            commands::assign::restart(ctx.primary_pool(), &sender, &deployment, sleep)
+            let pool = ctx.primary_pool();
+            commands::assign::restart(
+                ctx_clone.primary_pool(),
+                &sender,
+                &deployment.locate_unique(&pool).unwrap(),
+                sleep,
+            )
         }
```

Comment:
This is also needs similar changes like `Pause`

Reply:
It's done